spec/requests/users_controller_spec.rb

```diff
@@ -1876,9 +1876,27 @@ def enabled?
         SiteSetting.reserved_usernames = "reserved"
 
         put "/u/#{user.username}/preferences/username.json", params: { new_username: "reserved" }
-        body = response.parsed_body
 
-        expect(body["errors"].first).to include(I18n.t("login.reserved_username"))
+        expect(response.parsed_body["errors"].first).to include(I18n.t("login.reserved_username"))
+      end
+
+      it "allows admins to change a username to one in the reserved list" do
+        sign_in(admin)
+        SiteSetting.reserved_usernames = "reserved"
+
+        put "/u/#{user.username}/preferences/username.json", params: { new_username: "reserved" }
```

Comment:
```suggestion
        put "/u/#{user.username}/preferences/username.json", params: { new_username: "reserved" }
        expect(response.status).to eq(200)
```

Minor one but checking the response's status code will provide more useful error messages when the request fails for random reasons.

Reply:
Sure will change :+1: